conda_lockfiles/loaders/utils.py

```diff
@@ -0,0 +1,59 @@
+from typing import Any
+
+from conda.base.context import context
+from conda.core.package_cache_data import PackageCacheData, ProgressiveFetchExtract
+from conda.exceptions import (
+    CondaExitZero,
+    DryRunExit,
+)
+from conda.models.match_spec import MatchSpec
+from conda.models.records import PackageRecord
+
+CondaPackageURL = str
+CondaPackageMetadata = dict[str, Any]
+
+
+def _matchspec_from_url_and_metadata(
+    url: CondaPackageURL, metadata: CondaPackageMetadata
+):
+    checksums = {}
+    for checksum_name in ["md5", "sha256"]:
+        if checksum_name in metadata:
+            checksums[checksum_name] = metadata[checksum_name]
+    return MatchSpec(url, **checksums)
```

Comment:
If we instead do the following are we interested in inlining this below?

```suggestion
    return MatchSpec(
        url,
        **{
            key: metadata[key]
            for key in {"md5", "sha256"}.intersection(metadata)
        },
    )
```